File: Desafio Impar/vehiclecrm-backend/VehicleCRM/src/VehicleCRM.Domain/Entities/SaleOpportunity.cs
using VehicleCRM.Domain.Entities.Base;
using VehicleCRM.Domain.Enums;

namespace VehicleCRM.Domain.Entities
{
    public class SaleOpportunity : BaseEntity
    {
        protected SaleOpportunity() { }

        public SaleOpportunity(long customerId, long vehicleId, SaleOpportunityStatus status, decimal proposedValue, string negotiationNotes)
        {
            CustomerId = customerId;
            VehicleId = vehicleId;
            Status = status;
            ProposedValue = proposedValue;
            NegotiationNotes = negotiationNotes;
        }

        public virtual long CustomerId { get; private set; }
        public virtual long VehicleId { get; private set; }
        public virtual SaleOpportunityStatus Status { get; private set; }
        public virtual decimal ProposedValue { get; private set; }
        public virtual string NegotiationNotes { get; private set; }

        public virtual Customer Customer { get; private set; }
        public virtual Vehicle Vehicle { get; private set; }
    }
}

File: Desafio Impar/vehiclecrm-frontend/vehiclecrm/src/types/saleOpportunity.ts
export type SaleOpportunityStatus = 1 | 2 | 3 | 4 | 5

export interface SaleOpportunity {
  id: number
  proposedValue: number
  status: SaleOpportunityStatus
  notes?: string
  createDate: string
  modificationDate: string | null
  vehicle: {
    vehicleId: number
    model: string
  }
  customer: {
    customerId: number
    name: string
  }
}

export interface CreateSaleOpportunityDTO {
  customerId: number
  vehicleId: number
  status: SaleOpportunityStatus
  proposedValue: number
  notes?: string
}

export interface UpdateSaleOpportunityDTO {
  id: number
  customerId: number
  vehicleId: number
  status: SaleOpportunityStatus
  proposedValue: number
  notes?: string
}

export interface SaleOpportunityFilters {
  vehicleModel?: string
  customerName?: string
  proposedValueFrom?: number
  proposedValueTo?: number
  status?: SaleOpportunityStatus
}
